### ISSUE TITLE
JWE encryption is missing base64url-encoded protected header as AEAD

### ISSUE THREAD

**Issue by deeglaze** [confidential-containers/trustee#583]
### Describe the bug

The JWE protected_header is not protected AEAD as expected by the [RFC for JWE](https://datatracker.ietf.org/doc/html/rfc7516#section-3.3).

### How to reproduce

n/a

### CoCo version information

all

### What TEE are you seeing the problem on

Snp

**Comment by Xynnn007** [comment]
Yes. Currently we are using the function [encrypt](https://docs.rs/aes-gcm/0.10.3/aes_gcm/struct.AesGcm.html#method.encrypt) which will automatically append the auth `tag` to the returned ciphertext. To follow JWE RFC, we'd use [encrypt_in_place_detached](https://docs.rs/aes-gcm/0.10.3/aes_gcm/trait.AeadInPlace.html#tymethod.encrypt_in_place_detached) to explicitly mark the `tag` into responsed JWE.

**Comment by Xynnn007** [comment]
Ok. To follow rfc7516, we have the following roadmap
- [ ] https://github.com/virtee/kbs-types/pull/44
- [ ] https://github.com/confidential-containers/guest-components/pull/820
- [ ] https://github.com/confidential-containers/trustee/pull/597
